components/event-receiver/org.wso2.carbon.event.input.adapter.core/src/main/java/org/wso2/carbon/event/input/adapter/core/internal/InputAdapterRuntime.java

```diff
@@ -84,9 +84,13 @@ public void start() {
     public void destroy() {
         if (inputEventAdapter != null) {
             try {
-                inputEventAdapter.disconnect();
-            } finally {
-                inputEventAdapter.destroy();
+                try {
```

Comment:
@keizer619 
Thank you for pointing out. Fixed in https://github.com/wso2/carbon-analytics-common/pull/142